system/include/SSE/emmintrin.h

```diff
@@ -0,0 +1,1616 @@
+}
+
+static __inline__ int __attribute__((__always_inline__, __nodebug__))
+_mm_ucomineq_sd(__m128d __a, __m128d __b)
+{
+  return ((__f64x2)__a)[0] != ((__f64x2)__b)[0];
+}
+
+static __inline__ __m128 __attribute__((__always_inline__, __nodebug__))
+_mm_cvtpd_ps(__m128d __a)
+{
+  return (__m128)wasm_f32x4_make((float)__a[0], (float)__a[1], 0, 0);
+}
+
+static __inline__ __m128d __attribute__((__always_inline__, __nodebug__))
+_mm_cvtps_pd(__m128 __a)
+{
+  return (__m128d) { (double)__a[0], (double)__a[1] };
+}
+
+static __inline__ __m128d __attribute__((__always_inline__, __nodebug__))
+_mm_cvtepi32_pd(__m128i __a)
+{
+  return (__m128d) { (double)__a[0], (double)__a[1] };
+}
+
+static __inline__ __m128i __attribute__((__always_inline__, __nodebug__))
+_mm_cvtpd_epi32(__m128d __a)
+{
+  // TODO: OPTIMIZE!
+  int m[2];
+  for(int i = 0; i < 2; ++i)
+  {
+    int x = lrint(__a[i]);
+    if (x != 0 || fabs(__a[i]) < 2.0)
+      m[i] = (int)x;
+    else
+      m[i] = (int)0x80000000;
+  }
+  return (__m128i) { m[0], m[1], 0, 0 };
+}
+
+static __inline__ int __attribute__((__always_inline__, __nodebug__))
+_mm_cvtsd_si32(__m128d __a)
+{
+  // TODO: OPTIMIZE!
+  int x = lrint(__a[0]);
+  if (x != 0 || fabs(__a[0]) < 2.0)
+    return (int)x;
+  else
+    return (int)0x80000000;
+}
+
+static __inline__ __m128 __attribute__((__always_inline__, __nodebug__))
+_mm_cvtsd_ss(__m128 __a, __m128d __b)
+{
+  __a[0] = __b[0];
+  return __a;
+}
+
+static __inline__ __m128d __attribute__((__always_inline__, __nodebug__))
+_mm_cvtsi32_sd(__m128d __a, int __b)
+{
+  __a[0] = __b;
+  return __a;
+}
+
+static __inline__ __m128d __attribute__((__always_inline__, __nodebug__))
+_mm_cvtss_sd(__m128d __a, __m128 __b)
+{
+  __a[0] = __b[0];
+  return __a;
+}
+
+static __inline__ __m128i __attribute__((__always_inline__, __nodebug__))
+_mm_cvttpd_epi32(__m128d __a)
+{
+  // TODO: OPTIMIZE!
+  int m[2];
+  for(int i = 0; i < 2; ++i)
+  {
+    int x = lrint(__a[i]);
+    if (x != 0 || fabs(__a[i]) < 2.0)
+      m[i] = (int)__a[i];
+    else
+      m[i] = (int)0x80000000;
+  }
+  return (__m128i) { m[0], m[1], 0, 0 };
+}
+
+static __inline__ int __attribute__((__always_inline__, __nodebug__))
+_mm_cvttsd_si32(__m128d __a)
+{
+  // TODO: OPTIMIZE!
+  int x = lrint(__a[0]);
+  if (x != 0 || fabs(__a[0]) < 2.0)
+    return (int)__a[0];
+  else
+    return (int)0x80000000;
+}
+
+static __inline__ double __attribute__((__always_inline__, __nodebug__))
+_mm_cvtsd_f64(__m128d __a)
+{
+  return wasm_f64x2_extract_lane((v128_t)__a, 0);
+}
+
+static __inline__ __m128d __attribute__((__always_inline__, __nodebug__))
+_mm_load_pd(double const *__dp)
+{
+  return *(__m128d*)__dp;
+}
+
+static __inline__ __m128d __attribute__((__always_inline__, __nodebug__))
+_mm_load1_pd(double const *__dp)
+{
+  struct __mm_load1_pd_struct {
+    double __u;
+  } __attribute__((__packed__, __may_alias__));
+  double __u = ((struct __mm_load1_pd_struct*)__dp)->__u;
+  return (__m128d){ __u, __u };
```

Comment:
This looks like it could be implemented with a `load_splat` intrinsic.

Reply:
Check, converted to `wasm_v64x2_load_splat` in a followup PR.